compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -374,7 +374,20 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     |p| format!("`{}` defined here returns `{}`", p, callee_ty),
                                 )
                             }
-                            _ => Some(format!("`{}` defined here", callee_ty)),
+                            _ => {
+                                // If `def` is a local variable, it means that it is not a
+                                // definition itself, but a variable of that type, which means we
+                                // need to emit a different diagnostic.
```

Comment:
```suggestion
                                // definition itself but a variable of that type.
```

The action part is obvious, the explanation is good enough, I was confused reading the action.